### PR TITLE
Fix for #1528, setup ObjectMapper configuration for DgsWebSocketAutoC…

### PR DESCRIPTION
**Pull request by clayembry**
Fix for #1528, setup ObjectMapper configuration for DgsWebSocketAutoConfig to mirror DgsWebMvcAutoConfiguration.

Pull request checklist
----

- [x] Please read our [contributor guide](https://github.com/Netflix/dgs-framework/blob/master/CONTRIBUTING.md)
- [x] Consider creating a discussion on the [discussion forum](https://github.com/Netflix/dgs-framework/discussions)
  first
- [x] Make sure the PR doesn't introduce backward compatibility issues
- [x] Make sure to have sufficient test cases

Pull Request type
----

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe):

Changes in this PR
----

Issue #1528

The ObjectMappers in `WebsocketGraphQLTransportWSProtocolHandler` and `WebsocketGraphQLWSProtocolHandler` have been refactored for constructor injection instead of having each class instantiate its own ObjectMapper in a companion object. This fixes an immediate problem we are having where our subscriptions fail with errors because the subscription ObjectMappers do not have JSR310 time module configured. In addition, `DgsWebSocketAutoConfig` has been configured to create a Qualified ObjectMapper named "dgsObjectMapper" if it has not already been created.

My previous Kotlin and Gradle experience is limited, so I just followed the patterns for how ObjectMapper is used and configured in `DgsRestController` and `DgsWebMvcAutoConfiguration`. This PR establishes uniform ObjectMapper usage for graphql requests and subscriptions.

Alternatives considered
----

I could have added the JSR310 time module to the companion object ObjectMappers in `WebsocketGraphQLTransportWSProtocolHandler` and `WebsocketGraphQLWSProtocolHandler`. But then what happens the next time a change is needed to ObjectMapper configuration. It would have to be changed in many places instead of just updating the "dgsObjectMapper" config.

### CODE REVIEW

File: graphql-dgs-subscriptions-websockets-autoconfigure/src/test/kotlin/com/netflix/graphql/dgs/subscriptions/websockets/autoconfigure/DgsWebSocketAutoConfigTest.kt [hunk 1]
```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2023 Netflix, Inc.
```
Comment (srinivasankavitha) [inline graphql-dgs-subscriptions-websockets-autoconfigure/src/test/kotlin/com/netflix/graphql/dgs/subscriptions/websockets/autoconfigure/DgsWebSocketAutoConfigTest.kt:2]: You won't need this test once you get rid of the object mapper bean.

**Review by srinivasankavitha — CHANGES_REQUESTED** [review]
Thanks for the PR! The changes look good, have a few suggestions.

File: graphql-dgs-subscriptions-websockets-autoconfigure/src/main/kotlin/com/netflix/graphql/dgs/subscriptions/websockets/DgsWebSocketAutoConfig.kt [hunk 2]
```diff
@@ -34,8 +39,15 @@ import org.springframework.web.socket.server.support.DefaultHandshakeHandler
 @EnableConfigurationProperties(DgsWebSocketConfigurationProperties::class)
 open class DgsWebSocketAutoConfig {
     @Bean
```
[the remark this answers is not in the document]
Comment (clayembry) [reply graphql-dgs-subscriptions-websockets-autoconfigure/src/main/kotlin/com/netflix/graphql/dgs/subscriptions/websockets/DgsWebSocketAutoConfig.kt:41]: Yeah, I was thinking if someone was only using the subscription starter by itself. But probably that's not possible.

I'll remove the unnecessary bean config.

**Comment by srinivasankavitha** [comment]
You can fix lint failure by running `./gradlew formatKotlin`

**Review by srinivasankavitha — APPROVED** [review]

**Pull request merged**